DOC DBField setValue() changes (#720)
silverstripe/developer-docs: en/08_Changelogs/6.0.0.md

- Write a `BuildTask` that updates any invalid values in your database.
- While it's generally not recommended, you have the option of disabling validation via the [`DataObject.validation_enabled`](api:SilverStripe\ORM\DataObject->validation_enabled) configuration property.

> In addition, the following `DBField` subclasses have also had the follow changes made to them:

Reviewer: ```suggestion
In addition, the following `DBField` subclasses have also had the following changes made to them:
```

Author: Updated

> - [`DBBoolean`](api:SilverStripe\ORM\FieldType\DBBoolean) will no longer convert non-boolean like values to a `bool` when calling `setValue()`. For instance `"Amazing"` will no longer be converted to `true`, instead its value will remain as `"Amazing"` and it will throw a validation exception if attempted to be written to the database. There are some boolean-like values, such as `"t"`, that will continue to be converted to their boolean equivalents for backwards compatibility. See [validation added to `DBField`s](#dbfield-validation) for specifics.

Reviewer: ```suggestion
- [`DBBoolean`](api:SilverStripe\ORM\FieldType\DBBoolean) will no longer convert non-boolean like values to a `bool` when calling `setValue()`. For instance `"Amazing"` will no longer be converted to `true`, instead its value will remain as `"Amazing"` and it will throw a validation exception if attempted to be written to the database. There are some boolean-like values, such as `"t"`, that will continue to be converted to their boolean equivalents for backwards compatibility.
```
Now that it is in this section we don't need this link

Author: Updated

> - [`DBCurrency`](api:SilverStripe\ORM\FieldType\DBCurrency) will no longer parse numeric values contained in a string when calling `setValue()`. For instance "this is 50.29 dollars" will no longer be converted to "$50.29", instead its value will remain as "this is 50.29 dollars" and it will throw a validation exception if attempted to be written to the database.

Reviewer: ```suggestion
- [`DBCurrency`](api:SilverStripe\ORM\FieldType\DBCurrency) will no longer parse numeric values contained in a string when calling `setValue()`. For instance `"this is 50.29 dollars"` will no longer be converted to `"$50.29"`, instead its value will remain as `"this is 50.29 dollars"` and it will throw a validation exception if attempted to be written to the database.
```

Author: Updated

- `SilverStripe\ORM\DataExtension`, `SilverStripe\CMS\Model\SiteTreeExtension`, and `SilverStripe\Admin\LeftAndMainExtension` have been removed. If you subclass any of these classes, you must now subclass [`Extension`](api:SilverStripe\Core\Extension) directly instead.
- [`DBCurrency`](api:SilverStripe\ORM\FieldType\DBCurrency) will no longer parse numeric values contained in a string when calling `setValue()`. For instance "this is 50.29 dollars" will no longer be converted to "$50.29", instead its value will remain as "this is 50.29 dollars" and it will throw a validation exception if attempted to be written to the database.

> - [`DBDecimal`](api:SilverStripe\ORM\FieldType\DBDecimal) will no longer parse numeric values contained in a string when calling `setValue()`. For instance "This is -123.45" will no longer be converted to "-123.45", instead its value will remain as "This is -123.45" and it will throw a validation exception if attempted to be written to the database.

Reviewer: This could be misinterpreted to also mean `"-123.45"` won't be converted to `-123.45`.

```suggestion
- [`DBDecimal`](api:SilverStripe\ORM\FieldType\DBDecimal) will no longer parse numeric values contained in a string when calling `setValue()` unless the string is entirely numeric. For example `"This is -123.45"` will no longer be converted to `"-123.45"`, instead its value will remain as `"This is -123.45"` and it will throw a validation exception if attempted to be written to the database. Fully numeric strings like `"-123.45"` will still be cast and not throw validation exceptions.
```

Author: Updated

> - [`DBBoolean`](api:SilverStripe\ORM\FieldType\DBBoolean) will no longer convert non-boolean like values to a `bool` when calling `setValue()`. For instance "Amazing" will no longer be converted to `true`, instead its value will remain as "Amazing" and it will throw a validation exception if attempted to be written to the database. There are some boolean-like values, such as "t", that will continue to be converted to their boolean equivalents for backwards compatibility.

Reviewer: ```suggestion
- [`DBBoolean`](api:SilverStripe\ORM\FieldType\DBBoolean) will no longer convert non-boolean like values to a `bool` when calling `setValue()`. For instance `"Amazing"` will no longer be converted to `true`, instead its value will remain as `"Amazing"` and it will throw a validation exception if attempted to be written to the database. There are some boolean-like values, such as `"t"`, that will continue to be converted to their boolean equivalents for backwards compatibility. See [validation added to `DBField`s](#dbfield-validation) for specifics.
```

Author: Updated
